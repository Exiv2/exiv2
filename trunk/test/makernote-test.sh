#! /bin/sh
# Test driver for tests of MakerNoteFactory::match
results="./tmp/makernote-test.out"
good="./data/makernote-test.out"
(
binpath="../../src"
cd ./tmp
$binpath/makernote-test
) > $results

diff -q $results $good
rc=$?
if [ $rc -eq 0 ] ; then
    echo "All testcases passed."
else
    diff $results $good
fi
