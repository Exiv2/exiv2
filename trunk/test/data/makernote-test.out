Key 'Canon' matches 'Canon' with a score of 7.
Key 'Canon' matches 'Canon*' with a score of 6.
Key 'Canon Corp.' matches 'Canon*' with a score of 6.
Key 'PENTAX Corporation' matches 'PENTAX*' with a score of 7.
Key 'foobar' matches '*foo*bar*' with a score of 7.
Key 'barfoofoobarbar' matches '*foo*bar*' with a score of 7.
Key 'foo' does not match 'foo*bar'.
Key 'bar' does not match 'foo*bar'.
Key 'foobar' matches 'foo*bar' with a score of 7.
Key 'fooYAHOObar' matches 'foo*bar' with a score of 7.
Key 'Thefoobar' does not match 'foo*bar'.
Key 'foobarTrick' does not match 'foo*bar'.
Key 'ThefoobarTrick' does not match 'foo*bar'.
Key 'ThefooYAHOObarTrick' does not match 'foo*bar'.
Key 'anything' matches '*' with a score of 1.
Key 'anything times two' matches '**' with a score of 1.
Key 'bar' matches '*bar' with a score of 4.
Key 'bar' does not match 'b*bar'.
Key 'bbar' matches 'b*bar' with a score of 5.
Key 'bar' does not match '*foobar'.
Key 'foobar' matches '*bar' with a score of 4.
